cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/Filters/FilterCollection/FilterConditionInput/index.tsx

```diff
@@ -89,80 +90,86 @@ const FilterConditionInputview: React.FC<IFilterConditionInputProps> = ({
           Map({
             property: existingCondition.get('property') || '',
             operator: existingCondition.get('operator') || '',
-            value: existingCondition.get('value') || '',
+            ...(valueRequired && { value: existingCondition.get('value') || '' }),
           })
         )
       );
     }
-  }, [conditionMode]);
+  }, [conditionMode, valueRequired]);
 
   function setFilterCondition(conditionProperty: string) {
     return (val) => {
       setFilterToCondition(filterToCondition.setIn([filterID, conditionProperty], val));
     };
   }
 
-  return (
-    <If condition={!isNil(existingCondition)}>
-      <Heading type={HeadingTypes.h6} label="Show these widgets by the following condition..." />
-      <div className={classes.filterConditionInput}>
-        <PluginInput
-          widgetType={'radio-group'}
-          value={conditionMode}
-          onChange={setConditionMode}
-          label={'Condition Type'}
-          options={Object.values(FilterConditionMode).map((mode) => ({ id: mode, label: mode }))}
-          layout={'inline'}
-        />
-      </div>
-      <If condition={conditionMode === FilterConditionMode.Expression}>
-        <div className={classes.filterConditionInput}>
-          <PluginInput
-            widgetType={'textbox'}
-            value={existingCondition.get(FilterConditionProperty.Expression)}
-            onChange={setFilterCondition(FilterConditionProperty.Expression)}
-            label={FilterConditionProperty.Expression}
-            required={false}
-          />
-        </div>
-      </If>
-      <If condition={conditionMode === FilterConditionMode.Operator}>
+  return React.useMemo(
+    () => (
+      <If condition={!isNil(existingCondition)}>
```

Comment:
same here regarding `<If>`. Make sure you check the javascript console and there's no errors being thrown during runtime

Reply:
We have discussed and found out <If> can have multiple children.

Resolved.